cfme/cloud/stack.py

```diff
@@ -1,34 +1,254 @@
-from functools import partial
-from xml.sax.saxutils import quoteattr
-
 from navmazing import NavigateToSibling, NavigateToAttribute
-from selenium.common.exceptions import NoSuchElementException
-
-import cfme.fixtures.pytest_selenium as sel
-from cfme import web_ui as ui
+from widgetastic.widget import View
+from widgetastic.exceptions import NoSuchElementException
+from widgetastic_patternfly import Button, Dropdown, FlashMessages, BootstrapNav
+from widgetastic_manageiq import (
+    Accordion, BootstrapSelect, BreadCrumb, ItemsToolBarViewSelector, PaginationPane, Search,
+    SummaryTable, Table, Text)
+
+from cfme.base.ui import BaseLoggedInPage
 from cfme.exceptions import DestinationNotFound, StackNotFound, CandidateNotFound
-from cfme.web_ui import Quadicon, flash, Form, fill, form_buttons, paginator, toolbar as tb, \
-    match_location, accordion
-from cfme.exceptions import CFMEException, FlashMessageException
+from cfme.web_ui import match_location
+from cfme.exceptions import CFMEException
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import navigator, navigate_to, CFMENavigateStep
 from utils.pretty import Pretty
 from utils.wait import wait_for
-from utils.version import current_version
-
-
-cfg_btn = partial(tb.select, "Configuration")
-pol_btn = partial(tb.select, 'Policy')
-lifecycle_btn = partial(tb.select, 'Lifecycle')
 
-edit_tags_form = Form(
-    fields=[
-        ("select_tag", ui.Select("select#tag_cat")),
-        ("select_value", ui.Select("select#tag_add"))
-    ])
 
-match_page = partial(match_location, controller='orchestration_stack',
-                     title='Stacks')
+class StackToolbar(View):
+    """The toolbar on the stacks page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    lifecycle = Dropdown('Lifecycle')
+    download = Dropdown('Download')
+
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class StackDetailsToolbar(View):
+    """The toolbar on the stacks detail page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    lifecycle = Dropdown('Lifecycle')
+    download = Button('Download summary in PDF format')
+
+
+class StackSubpageToolbar(View):
+    """The toolbar on the sub pages, like resources and security groups"""
+    show_summary = Button('Show {} Summary')      # TODO How to get name in there?
```

Comment:
Just looking at the code for ParametrizedString (the base class for Parameter), it doesn't look like it supports the parameter format I would need. In this particular case I want the "name" attribute of the "object" object within the context.

I usually use this when checking the page title:

```python
expected_title = '{} (Summary)'.format(self.context['object'].name)
```

However, Parameter only seems to support objects within the context object. From widgetastic.utils:

```python
param_value = view.context[context_name]
```

This means that what I would _want_ to do is this:

```python
show_summary = Button(Parameter('Show {object.name} Summary'))
```

But that would result in the following lookup, which wouldn't work:

```python
param_value = view.context['object.name']
```